### PR TITLE
Baidu Schorlar: Fix url and scrape more data

### PR DESCRIPTION
**Pull request by kun-g**
Can scrape multiple items now

### CODE REVIEW

File: Baidu Scholar.js [hunk 1]
```diff
@@ -106,16 +110,63 @@ function scrape(doc, url) {
...
+			// })
+			item.tags = tags
+			if (item.title == null) {
+				item.title = title
+			}
+			if (!item.creators || item.creators.length == 0) {
+				item.creators = []
+				doc.querySelectorAll('p.author_text a').forEach(e => {
+					var name = ZU.trimInternal(e.textContent)
+					var index = name.indexOf(',')
```
Comment (adam3smith) [inline Baidu Scholar.js]: This is built in as `ZU.cleanAuthor` See https://www.zotero.org/support/dev/translators/coding#utility_functions

File: Baidu Scholar.js [hunk 2]
```diff
@@ -106,16 +110,63 @@ function scrape(doc, url) {
...
+					var lastName = name.slice(0, index)
+					var firstName = name.slice(index+1)
+					item.creators.push({ lastName, firstName, creatorType: 'author' })
+				})
+			}
+			if (!item.publicationTitle && attr(doc, 'a.journal_title', 'title')) {
```
Comment (adam3smith) [inline Baidu Scholar.js]: no need to test for the `attr` part. If it's empty, the publication title will just remain empty.

File: Baidu Scholar.js [hunk 3]
```diff
@@ -106,16 +110,63 @@ function scrape(doc, url) {
...
+				item.publicationTitle = attr(doc, 'a.journal_title', 'title')
+			}
+			if (!item.date && text(doc, 'div.year_wr p.kw_main')) {
+				item.date = ZU.trimInternal(text(doc, 'div.year_wr p.kw_main'))
+			}
+			if (item.DOI == null && text(doc, 'div.doi_wr p.kw_main')) {
```
Comment (adam3smith) [inline Baidu Scholar.js]: ditto here. Also use the same `!item.DOI` syntax as for the other tests.

File: Baidu Scholar.js [hunk 4]
```diff
@@ -106,16 +110,63 @@ function scrape(doc, url) {
...
+				item.DOI = ZU.trimInternal(text(doc, 'div.doi_wr p.kw_main'))
+			}
+			if (!item.extra && text(doc, 'p.ref-wr-num a')) {
+				let n = ZU.trimInternal(text(doc, 'p.ref-wr-num a'))
+				item.extra = "ZSCC:"+Array(7-n.length).fill('0').join('')+n
```
Comment (adam3smith) [inline Baidu Scholar.js]: What is ZSCC?

File: Baidu Scholar.js [hunk 5]
```diff
@@ -106,16 +110,63 @@ function scrape(doc, url) {
...
 			}
 			item.attachments.push({
-				title: "Snapshot",
-				document: doc
+				url,
+				title: "Baidu学术地址",
+				mimeType: "text/html",
+				snapshot: false,
 			});
+			// Sources
```
Comment (adam3smith) [inline Baidu Scholar.js]: I don't think sources and related do what you think they do -- they're not realy usable in web translators, so please delete this code.

File: Baidu Scholar.js [hunk 6]
```diff
@@ -87,12 +87,16 @@ function scrape(doc, url) {
 	var diversion = attr(doc, 'i.reqdata', 'diversion');
 	var sign = attr(doc, 'a.sc_q' , 'data-sign');
 	var risUrl = "http://xueshu.baidu.com/u/citation?&url=" + encodeURIComponent(dataUrl) + "&sign=" + sign + "&diversion=" + diversion + "&t=ris";
+	var title = doc.title.replace('_百度学术', '')
```
Comment (adam3smith) [inline Baidu Scholar.js]: is this used anywhere? If not, delete

File: Baidu Scholar.js [hunk 7]
```diff
@@ -106,16 +110,63 @@ function scrape(doc, url) {
 				item.abstractNote = text(doc, 'div.sc_abstract') || text(doc, 'p.abstract');
 			}
 			item.attachments.push({
```
Comment (adam3smith) [inline Baidu Scholar.js]: what's wrong with the original here? We'd definitely want "Snapshot" to be part of the title to ensure uniform behavior across Zotero.

File: Baidu Scholar.js (hunk 4 above)
Comment (kun-g) [reply Baidu Scholar.js]: It is from [zotero-scholar-citations](https://github.com/MaxKuehn/zotero-scholar-citations). Thought it was standard. Should I leave it empty?

File: Baidu Scholar.js (hunk 6 above)
Comment (kun-g) [reply Baidu Scholar.js]: There are cases RIS file is invalid, this is an insurance.

File: Baidu Scholar.js (hunk 7 above)
Comment (kun-g) [reply Baidu Scholar.js]: OK

File: Baidu Scholar.js (hunk 4 above)
Comment (adam3smith) [reply Baidu Scholar.js]: Yes, let's remove this please and update tests accordingly

File: Baidu Scholar.js [hunk 8]
```diff
@@ -87,12 +87,16 @@ function scrape(doc, url) {
 	var diversion = attr(doc, 'i.reqdata', 'diversion');
 	var sign = attr(doc, 'a.sc_q' , 'data-sign');
 	var risUrl = "http://xueshu.baidu.com/u/citation?&url=" + encodeURIComponent(dataUrl) + "&sign=" + sign + "&diversion=" + diversion + "&t=ris";
+	var title = doc.title.replace('_百度学术', '')
+
+	var tags = []
+	doc.querySelectorAll('p.kw_main span a').forEach(e => tags.push(ZU.trimInternal(e.textContent)))
```
Comment (zuphilip) [inline Baidu Scholar.js]: Does that work at all?

Comment (zuphilip) [reply Baidu Scholar.js]: NVM I see now that it is okay.

**Comment by zuphilip** [comment]
I fixed the linting issues (e.g. we always want a semi-colon at the end of the line) and squashed the commits logically together.

**Comment by zuphilip** [comment]
Merged :rocket: ! Thank you very much @kun-g for all the work! (You may want to clean your master branch now and use a different one for future PR.)

**Pull request merged**